code/modules/food_and_drinks/plate.dm

```diff
@@ -0,0 +1,63 @@
+/obj/item/plate
+	name = "plate"
+	desc = "Holds food, powerful. Good for morale when you're not eating your spaghetti off of a desk."
+	icon = 'icons/obj/kitchen.dmi'
+	icon_state = "plate"
+	w_class = WEIGHT_CLASS_HUGE //No backpack.
+	///How many things fit on this plate?
+	var/max_items = 8
+	///The offset from side to side the food items can have on the plate
+	var/max_x_offset = 4
+	///The max height offset the food can reach on the plate
+	var/max_height_offset = 5
+	///Offset of where the click is calculated from, due to how food is positioned in their DMIs.
+	var/placement_offset = -12
+
+
+/obj/item/plate/attackby(obj/item/I, mob/user, params)
+	if(!IS_EDIBLE(I))
+		to_chat(user, "<span class='notice'>[src] is made for food, and food alone!</span>")
+		return
+	if(contents.len >= max_items)
+		to_chat(user, "<span class='notice'>[src] can't fit more items!</span>")
+		return
+	var/list/modifiers = params2list(params)
+	//Center the icon where the user clicked.
+	if(!LAZYACCESS(modifiers, ICON_X) || !LAZYACCESS(modifiers, ICON_Y))
+		return
+	if(user.transferItemToLoc(I, src, silent = FALSE))
+		I.pixel_x = clamp(text2num(LAZYACCESS(modifiers, ICON_X)) - 16, -max_x_offset, max_x_offset)
+		I.pixel_y = min(text2num(LAZYACCESS(modifiers, ICON_Y)) + placement_offset, max_height_offset)
+		to_chat(user, "<span class='notice'>You place [I] on [src].</span>")
+		AddToPlate(I, user)
+		update_appearance()
+	else
+		return ..()
+
+/obj/item/plate/pre_attack(atom/A, mob/living/user, params)
+	if(!iscarbon(A))
+		return
+	if(!contents.len)
+		return
+	var/obj/item/object_to_eat = contents[1]
+	A.attackby(object_to_eat, user)
+	return TRUE //No normal attack
+
+///This proc adds the food to viscontents and makes sure it can deregister if this changes.
+/obj/item/plate/proc/AddToPlate(obj/item/item_to_plate, mob/user)
+	vis_contents += item_to_plate
+	item_to_plate.flags_1 |= IS_ONTOP_1
+	RegisterSignal(item_to_plate, COMSIG_MOVABLE_MOVED, .proc/ItemMoved)
+	RegisterSignal(item_to_plate, COMSIG_PARENT_QDELETING, .proc/ItemMoved)
+
+///This proc cleans up any signals on the item when it is removed from a plate, and ensures it has the correct state again.
+/obj/item/plate/proc/ItemRemovedFromPlate(obj/item/removed_item)
```

Comment:
uh... how DO you remove an item from a plate? if you put a piece of food on a plate, is it just trapped there until you eat it?

Reply:
you click it, the item is just on the plate like on a griddle or on a table.